tests/dialects/test_tsql.py

```diff
@@ -818,8 +818,23 @@ def test_add_date(self):
         )
 
     def test_date_diff(self):
-        self.validate_identity("SELECT DATEDIFF(year, '2020/01/01', '2021/01/01')")
-
+        self.validate_identity("SELECT DATEDIFF(year, '2020-01-01', '2021/01/01')")
+        self.validate_all(
+            "SELECT DATEDIFF(quarter, 0, '2021-01-01')",
+            write={
+                "tsql": "SELECT DATEDIFF(quarter, '1900-01-01', '2021-01-01')",
+                "spark": "SELECT DATEDIFF(quarter, '1900-01-01', '2021-01-01')",
+                # "duckdb": "SELECT DATEDIFF('quarter', TIMESTAMP'1900-01-01', TIMESTAMP'2021-01-01')",
```

Comment:
You can use `DateStrToDate` or perhaps the broader `TimeStrToTime` to represent T-SQL's `DATEDIFF` arguments at the AST level. That way you'll automatically cover these DuckDB test cases.